### PR TITLE
DM-41049: Add matched color difference plots/metrics

### PR DESCRIPTION
**Pull request by taranu**

### CODE REVIEW

**Review by natelust — CHANGES_REQUESTED** [review]
There are lots of random little comments, but I am happy for you to use your judgement on most of this. The only real place I would like changes, or at least explanation is where you are introducing the dict types that need changes in pex_config. I think there are other ways of doing that as I mentioned and I would rather see them done that way instead of changing pex_config.

File: python/lsst/analysis/tools/atools/diffMatched.py [hunk 1]
```diff
@@ -23,27 +23,132 @@
...
+        for action in self.actions.values():
+            result = action(data=data, **kwargs)
+            results.update(result)
+        return results
+
+
+class DictPlotAction(PlotAction):
+    """A dictionary of named PlotActions."""
+
+    actions = ConfigDictField[str, PlotAction](doc="Named plot actions")
```
[the remark this answers is not in the document]
Comment (taranu) [reply python/lsst/analysis/tools/atools/diffMatched.py:91]: I do need to add PlotActions dynamically based on the config (specified colours). I figured a generic `DictPlotAction` would be useful (and should go somewhere else) and simpler/better than overriding produce, no?

[the remark this answers is not in the document]
Comment (taranu) [reply python/lsst/analysis/tools/atools/diffMatched.py:91]: If `actions` can be trivially replaced with a `ConfigurableActionStructField`, fine by me, though at some point I'd like to understand how/why it supports derived types while regular `ConfigDictField` doesn't.

File: python/lsst/analysis/tools/actions/vector/vectorActions.py [hunk 2]
```diff
@@ -134,6 +136,57 @@ def __call__(self, data: KeyedData, **kwargs) -> Vector:
...
 
 
+class ColorDiff(VectorAction):
+    """Calculate the difference between two colors from flux actions."""
+
+    color1_flux1 = ConfigurableActionField[VectorAction](doc="Action providing first color's first flux")
+    color1_flux2 = ConfigurableActionField[VectorAction](doc="Action providing first color's second flux")
+    color2_flux1 = ConfigurableActionField[VectorAction](doc="Action providing second color's first flux")
+    color2_flux2 = ConfigurableActionField[VectorAction](doc="Action providing second color's second flux")
+    returnMillimags = Field[bool](doc="Use millimags or not?", default=True)
```
[the remark this answers is not in the document]
Comment (taranu) [reply python/lsst/analysis/tools/actions/vector/vectorActions.py:146]: It's copypasta from `MagDiff` which dates back to `analysis_drp`, but sure, now is as good a time as any to fix them.

File: python/lsst/analysis/tools/atools/genericProduce.py [hunk 3]
```diff
@@ -30,6 +32,11 @@
 class MagnitudeScatterPlot(MagnitudeXTool):
     """A scatter plot with a magnitude on the x-axis."""
 
+    suffixes_y_finalize = ListField[str](
+        doc="Suffixes for y-axis keys to finalize summary stats for",
```
[the remark this answers is not in the document]
Comment (taranu) [reply python/lsst/analysis/tools/atools/genericProduce.py:36]: No, that's it... but I'll re-word it.

File: python/lsst/analysis/tools/atools/genericProduce.py [hunk 4]
```diff
@@ -79,11 +86,16 @@ def finalize(self):
...
+                statAction.highSNSelector.fluxType = fluxType
+                statAction.highSNSelector.threshold = 200
+                statAction.lowSNSelector.fluxType = fluxType
+                statAction.lowSNSelector.threshold = 10
+                statAction.fluxType = fluxType
+                # TODO: it should be possible to name this something else...
+                # ... but ScatterPlotStatsAction uses self.identity for names
+                # And then ScatterPlotWithTwoHists relies on self.identity
+                # being Stars/Galaxies
+                setattr(self.process.calculateActions, plural, statAction)
```
[the remark this answers is not in the document]
Comment (taranu) [reply python/lsst/analysis/tools/atools/genericProduce.py:101]: Whoops... good point.

**Review by natelust — APPROVED** [review]

**Pull request merged**